lib/otto_api/site.ex

```diff
@@ -0,0 +1,30 @@
+defmodule OttoApi.Site do
+  @enforce_keys [:id, :url, :account_id, :inserted_at]
+  defstruct @enforce_keys
+
+  alias OttoApi.Client
+
+  @spec all(client :: %Client{}) ::
+          {:ok,
+           list(%__MODULE__{id: binary, url: binary, account_id: binary, inserted_at: binary})}
+  def all(client) do
+    {:ok,
+     %{
+       "data" => records
+     }} = Client.get(client, "/sites")
+
+    accounts =
+      Enum.map(records, fn record ->
+        %{"id" => id, "url" => url, "account_id" => account_id, "inserted_at" => inserted_at} =
+          record
+
+        %__MODULE__{id: id, url: url, account_id: account_id, inserted_at: inserted_at}
+      end)
+
+    {:ok, accounts}
+  end
+
+  def create(client, site) do
+    {:ok, response} = Client.post(client, "/sites", %{"site" => Map.from_struct(site)})
```

Comment:
If you want to silence the warning but still show response this can be changed to something like this:
```
{:ok, _response} = Client.post(client, "/sites", %{"site" => Map.from_struct(site)})
```

The main issue with this is that if `Client.post/3` returns an error the `:ok` tuple won't match and it will crash.

Reply:
Cool.  Ya I've got proper error handling on my list of things to figure out.  If you wanted to do one example of the way you'd do it I will copy.  I assume I'd just do case statements where it makes sense.  For now I'm just going with crashing.

Reply:
I would probably do this using `with` like this:
```
with {:ok, response} <- Client.post(client, "/sites", %{"site" => Map.from_struct(site)}),
     {:ok, site_struct} <- helper_site_from_create_resp(response)
  do
    {:ok, site_struct}
  else
    {:error, :reason_one} -> {:error, "some reason"}
    {:error, :reason_two} -> {:error, "some other reason"}
    :error -> {:error, "not sure"}
    other -> {:error, "who knows"}
end
```

It's also possible to represent this state machine as a set of functions, one for each state. This is what I used to do before `with` was introduced to the language. Sometimes I'll still use helper functions with some of the steps if I need the error `reason` values formatted in a specific way to make them useful. If `:error` can indicate more than one error state it's not that useful as a response value.
 
I would also suggest accepting `params` as a `Map.t()` in functions like this that create a new record and returning the struct when the record has been successfully created.